src/VisualStudio/Core/Def/Implementation/Experimentation/KeybindingResetDetector.cs

```diff
@@ -0,0 +1,340 @@
+                              action: OpenExtensionsHyperlink),
+                new InfoBarUI(title: ServicesVSResources.Never_show_this_again,
+                              kind: InfoBarUI.UIKind.Button,
+                              action: NeverShowAgain),
+                new InfoBarUI(title: "", kind: InfoBarUI.UIKind.Close,
+                              action: InfoBarClose));
+        }
+
+        private void RestoreVsKeybindings()
+        {
+            AssertIsForeground();
+
+            if (_uiShell == null)
+            {
+                _uiShell = _serviceProvider.GetService(typeof(SVsUIShell)) as IVsUIShell;
+            }
+
+            ErrorHandler.ThrowOnFailure(_uiShell.PostExecCommand(
+                    VSConstants.GUID_VSStandardCommandSet97,
+                    (uint)VSConstants.VSStd97CmdID.CustomizeKeyboard,
+                    (uint)OLECMDEXECOPT.OLECMDEXECOPT_DODEFAULT,
+                    null));
+
+            KeybindingsResetLogger.Log("KeybindingsReset");
+
+            _workspace.Options = _workspace.Options.WithChangedOption(KeybindingResetOptions.NeedsReset, false);
+        }
+
+        private void OpenExtensionsHyperlink()
+        {
+            ThisCanBeCalledOnAnyThread();
+            Process.Start(KeybindingsFwLink);
+
+            KeybindingsResetLogger.Log("ExtensionsLink");
+            _workspace.Options = _workspace.Options.WithChangedOption(KeybindingResetOptions.NeedsReset, false);
+        }
+
+        private void NeverShowAgain()
+        {
+            AssertIsForeground();
+
+            _workspace.Options = _workspace.Options.WithChangedOption(KeybindingResetOptions.NeverShowAgain, true);
+            KeybindingsResetLogger.Log("NeverShowAgain");
+
+            // The only external references to this object are as callbacks, which are removed by the dispose method.
+            Dispose();
+        }
+
+        private void InfoBarClose()
+        {
+            _infoBarOpen = InfoBarClosed;
+        }
+
+        public int QueryStatus(ref Guid pguidCmdGroup, uint cCmds, OLECMD[] prgCmds, IntPtr pCmdText)
+        {
+            // We don't care about query status, only when the command is actually executed
+            return (int)OLE.Interop.Constants.OLECMDERR_E_NOTSUPPORTED;
+        }
+
+        public int Exec(ref Guid pguidCmdGroup, uint nCmdID, uint nCmdexecopt, IntPtr pvaIn, IntPtr pvaOut)
+        {
+            if (pguidCmdGroup == ReSharperCommandGroup && nCmdID >= ResumeId && nCmdID <= ToggleSuspendId)
+            {
+                // Don't delay command processing to update resharper status
+                Task.Run(() => InvokeBelowInputPriority(UpdateReSharperEnableStatus));
+            }
+
+            // No matter the command, we never actually want to respond to it, so always return not supported. We're just monitoring.
+            return (int)OLE.Interop.Constants.OLECMDERR_E_NOTSUPPORTED;
+        }
+
+
+        void Dispose(bool disposing)
```

Comment:
missing accessibility keyword

Reply:
This was generated by our quickfix. Perhaps we should update the template to have the correct modifiers.

Reply:
We need to update it to obey style settings at some point - once we have the analyzer to require them (which I think we have part of now).